### PR TITLE
Errors have button to get Alert URL

### PR DESCRIPTION
**Pull request by hyu**
Hidden under a feature flag
(Add `enable_feature("team_alerts")` to your Tiltfile to see this WIP feature)

We were using `titleMsg` and `header` for the little bit of additional meta text we send along with errors. I standardized to the latter.

2019-07-26 - Alert button to get link

### CODE REVIEW

File: internal/hud/server/server.go [hunk 1]
```diff
@@ -10,15 +10,20 @@ import (
 	_ "github.com/gorilla/websocket"
 	"github.com/windmilleng/wmclient/pkg/analytics"
 
+	"context"
```
Comment (jazzdan) [inline internal/hud/server/server.go]: This should move up with the other built in packages

File: web/src/HUD.tsx [hunk 2]
```diff
@@ -130,20 +137,51 @@ class HUD extends Component<HudProps, HudState> {
...
+  }
+
   render() {
     let view = this.state.View
     let sailEnabled = view && view.SailEnabled ? view.SailEnabled : false
     let sailUrl = view && view.SailURL ? view.SailURL : ""
     let needsNudge = view ? view.NeedsAnalyticsNudge : false
     let message = this.state.Message
     let resources = (view && view.Resources) || []
+    let featureFlags = (view && view.FeatureFlags) || {}
```
Comment (jazzdan) [inline web/src/HUD.tsx]: I think this variable is unused?

File: web/src/__snapshots__/AlertPane.test.tsx.snap [hunk 3]
```diff
@@ -15,14 +15,14 @@ exports[`renders multiple lines of a crash log 1`] = `
         <p>
           Pod crashed
         </p>
-        <p>
+        <time>
```
Comment (landism) [inline web/src/__snapshots__/AlertPane.test.tsx.snap]: `<time>` inside `<time>` seems weird, as does a `<time>` with no `dateTime` attribute. Can we just get rid of the outer one? Was the `<p>` screwing things up?

File: internal/tft/client/client.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("Expected 200 response code from backend, got %d", resp.StatusCode)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	fmt.Println(string(body))
```
Comment (landism) [inline internal/tft/client/client.go]: 👀

File: internal/hud/server/server.go [hunk 5]
```diff
@@ -219,3 +227,62 @@ func MaybeSendToTriggerQueue(st store.RStore, name string) error {
...
+
+	decoder := json.NewDecoder(req.Body)
+	var alert tsAlert
+	err := decoder.Decode(&alert)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error decoding request: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	ctx := context.TODO()
```
[the remark this answers is not in the document]
Comment (jazzdan) [reply internal/hud/server/server.go]: I don't think so, unless we make context part of the wire tree which seems weird? We should definitely have a timeout on this though, I'll do that.

File: web/src/__snapshots__/AlertPane.test.tsx.snap (hunk 3 above)
Comment (hyu) [reply web/src/__snapshots__/AlertPane.test.tsx.snap]: Good catch! When we added relative time we neglected to remove this wrapping HTML.

**Pull request merged**